src/hotspot/os/aix/os_aix.cpp

```diff
@@ -1489,14 +1495,14 @@ struct vmembk_t {
   // also check that range is fully page aligned to the page size if the block.
   void assert_is_valid_subrange(char* p, size_t s) const {
     if (!contains_range(p, s)) {
-      trcVerbose("[" PTR_FORMAT " - " PTR_FORMAT "] is not a sub "
-              "range of [" PTR_FORMAT " - " PTR_FORMAT "].",
-              p2i(p), p2i(p + s), p2i(addr), p2i(addr + size));
-      guarantee0(false);
+      fatal(os,map)(RANGEFMT " is not a sub "
+              "range of " RANGEFMT, RANGEFMTARGS(p, s),
+              RANGEFMTARGS(addr, size));
     }
     if (!is_aligned_to(p, pagesize) || !is_aligned_to(p + s, pagesize)) {
-      trcVerbose("range [" PTR_FORMAT " - " PTR_FORMAT "] is not"
-              " aligned to pagesize (%lu)", p2i(p), p2i(p + s), (unsigned long) pagesize);
+      log_trace(os, map)("range " RANGEFMT " is not aligned to pagesize (%lu)",
+                         RANGEFMTARGS(p, s),
+                         (unsigned long)pagesize);
       guarantee0(false);
```

Comment:
I think the same thing can be done here - move the message into the fatal message.  Why does AIX have this?

#define assert0(b) assert((b), "")
#define guarantee0(b) guarantee((b), "")

??

Reply:
+1 to fatal().

>Why does AIX have this?
>
>#define assert0(b) assert((b), "")
>#define guarantee0(b) guarantee((b), "")
>
>??

This is my code I think, from way back when we started the AIX port at SAP. Based on Suns licensed hotspot sources. I really disliked having to state a reason for the assert every time I wrote one, so I saved me work by adding the 0 variants. I kind of came to terms with the reason parameter now.

Note that was from a time where we could not give back coding upstream, so the AIX port has a number of constructs like this.

Reply:
But lets leave this for now; lets limit this PR to logging. Code overhaul of the AIX os layer can be done separately.

Reply:
Thanks for the explanation of assert0 and guarantee0.  I generally find the reason parameter serves the same purpose as a comment except in some places where it's simply "sanity", but I get it.